### PR TITLE
Feat: User Detail Added

### PR DESCRIPTION
**Pull request by aliaankhowaja**
feat: added User detail endpoint

### CODE REVIEW

**Comment by Ali-Salman29** [comment]
Update the title of the PR.

File: app/core/api/v1/views/user.py [hunk 1]
```diff
@@ -17,3 +18,10 @@ class PersonRetrieveView(RetrieveAPIView):
 
     def get_object(self):
         return self.request.user.person
+
+class UserRetrieveView(RetrieveAPIView):
+    permission_classes = [IsAuthenticated]
+    serializer_class = UserDetailSerializer
+    def get_object(self):
+        person = Person.objects.get(user = self.request.user)
+        return person
```
Comment (Ali-Salman29) [inline app/core/api/v1/views/user.py]: Add the extra line at the end of the file.

Comment (Ali-Salman29) [reply app/core/api/v1/views/user.py]: You can use Pylint for this or a black formatter.

File: app/core/api/serializers/user.py [hunk 2]
```diff
@@ -7,3 +7,13 @@ class PersonSerializer(serializers.ModelSerializer):
     class Meta:
         model = Person
         fields = ['id', 'username', 'avatar', 'full_name', 'is_active']
+
+class UserDetailSerializer(serializers.ModelSerializer):
```
Comment (salman299) [inline app/core/api/serializers/user.py]: Use one serializer only.

Comment (salman299) [reply app/core/api/serializers/user.py]: Also, add manual fields such as first_name, father_name, etc in the serializer. There are some fields like id, created_date, and updated_date that are set automatically.

File: app/core/api/v1/views/user.py [hunk 3]
```diff
@@ -17,3 +18,10 @@ class PersonRetrieveView(RetrieveAPIView):
 
     def get_object(self):
         return self.request.user.person
+
+class UserRetrieveView(RetrieveAPIView):
```
Comment (salman299) [inline app/core/api/v1/views/user.py]: You can use a single view for updating and retrieving by overriding the get and put methods.

File: app/core/api/v1/views/user.py [hunk 4]
```diff
@@ -17,3 +18,10 @@ class PersonRetrieveView(RetrieveAPIView):
...
+    permission_classes = [IsAuthenticated]
+    serializer_class = UserDetailSerializer
+    def get_object(self):
+        person = Person.objects.get(id = self.request.user.id)
+        return person
```
Comment (salman299) [inline app/core/api/v1/views/user.py]: Add the extra space, or use pylint or black formatter.

**Review by salman299 — CHANGES_REQUESTED** [review]

File: app/core/api/v1/views/user.py (hunk 3 above)
Comment (Ali-Aan-Khowaja) [reply app/core/api/v1/views/user.py:22]: Person Retrieve view was created by you. Have changed the UserRetrieveView to UserRetrieveUpdateView

**Review by Ali-Salman29 — CHANGES_REQUESTED** [review]

File: app/core/api/serializers/user.py [hunk 5]
```diff
@@ -1,9 +1,41 @@
...
-    username = serializers.CharField(source='user.username', read_only=True)
+    username = serializers.CharField(source="user.username", read_only=True)
+
+    class Meta:
+        model = Person
+        fields = ["id", "username", "avatar", "full_name", "is_active"]
+
 
+class UserDetailUpdateSerializer(serializers.ModelSerializer):
     class Meta:
```
Comment (salman299) [inline app/core/api/serializers/user.py:14]: Add read_only_fields = ["created_at", "modified_at", "person_id"] and remove is_loggable and user from the fields list.

File: app/core/api/v1/views/user.py [hunk 6]
```diff
@@ -1,19 +1,41 @@
...
 class PersonRetrieveView(RetrieveAPIView):
     permission_classes = [IsAuthenticated]
     serializer_class = PersonSerializer
 
     def get_object(self):
         return self.request.user.person
+
+
+class UserRetrieveUpdateView(RetrieveUpdateAPIView):
+    permission_classes = [IsAuthenticated]
```
Comment (salman299) [inline app/core/api/v1/views/user.py:30]: Add a permission that only the user can update its information.

File: app/core/api/v1/views/user.py [hunk 7]
```diff
@@ -1,19 +1,41 @@
...
+    ListAPIView,
+    RetrieveAPIView,
+    UpdateAPIView,
+    RetrieveUpdateAPIView,
+)
 from rest_framework.filters import SearchFilter
 from rest_framework.permissions import IsAdminUser, IsAuthenticated
+from app.core.api.serializers.user import (
+    UserDetailUpdateSerializer,
+    # UserUpdateSerializer,
```
Comment (salman299) [inline app/core/api/v1/views/user.py]: Remove unnecessary imports

**Review by salman299 — CHANGES_REQUESTED** [review]

**Review by salman299 — APPROVED** [review]

**Review by Ali-Salman29 — APPROVED** [review]

**Pull request merged**